adal/authentication_context.py

```diff
@@ -75,11 +75,19 @@ def __init__(
             read timeout) <timeouts>` tuple.
         :param enable_pii: (optional) Unless this is set to True,
             there will be no Personally Identifiable Information (PII) written in log.
+        :param verify_ssl: (optional) requests verify. Either a boolean, in which case it 
+            controls whether we verify the server's TLS certificate, or a string, in which 
+            case it must be a path to a CA bundle to use. This value overrides env 
+            variable ADAL_PYTHON_SSL_NO_VERIFY.
+        :param proxies: (optional) requests proxies. Dictionary mapping protocol to the URL 
+            of the proxy. See http://docs.python-requests.org/en/master/user/advanced/#proxies
+            for details.
         '''
         self.authority = Authority(authority, validate_authority is None or validate_authority)
         self._oauth2client = None
         self.correlation_id = None
-        env_value = os.environ.get('ADAL_PYTHON_SSL_NO_VERIFY')
+        env_verify = False if 'ADAL_PYTHON_SSL_NO_VERIFY' in os.environ else None

```

Comment:
How about `env_verify = 'ADAL_PYTHON_SSL_NO_VERIFY' not in os.version`? Because `None` is conceptually NOT an expected value by [requests' verify](http://docs.python-requests.org/en/master/user/advanced/#ssl-cert-verification) anyway.

Reply:
`None` is the actual default value of requests actually, if you look at the method signature:
https://github.com/requests/requests/blob/1e8a3142d2376e86d8f923840c0b8d361af7786b/requests/sessions.py#L449

This is done that way in the operation signature, so a not-None value override the `verify` static value in the `Session` object (which this one is `True` by default: https://github.com/requests/requests/blob/1e8a3142d2376e86d8f923840c0b8d361af7786b/requests/sessions.py#L373). Since you don't use a `Session`, that would be equivalent, but `None` is accurate as well in this context. As long as you don't decide to use a common `Session` in the context, our two codes are indeed equivalent :).
Tell me what you prefer